src/dotty/tools/backend/jvm/CollectSuperCalls.scala

```diff
@@ -14,17 +14,18 @@ import dotty.tools.dotc.transform.TreeTransforms.{MiniPhaseTransform, Transforme
  *  the redundant mixin class could be required as a parent by the JVM.
  */
 class CollectSuperCalls extends MiniPhaseTransform {
+  import tpd._
 
   def phaseName: String = "collectSuperCalls"
 
-  override def transformSuper(tree: Super)(implicit ctx: Context, info: TransformerInfo): Tree = {
+  override def transformSelect(tree: Select)(implicit ctx: Context, info: TransformerInfo): Tree = {
     tree match {
-      case Trees.Super(qual: This, mix) if mix.nonEmpty =>
```

Comment:
Fruitless outer pattern match (and this matters at least a little bit, because Select is a common node). More efficient:

    tree.qualifier match { ... }